### PR TITLE
Add types for @fusebit/add-on-sdk

### PR DESCRIPTION
**Pull request by andrewrmiller**
This change adds types for the core functionality in [@fusebit/add-on-sdk](https://www.npmjs.com/package/@fusebit/add-on-sdk).  The type information provided here allows for basic interaction with the Fusebit platform (https://fusebit.io and https://developer.fusebit.io/docs)

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## fusebit__add-on-sdk ([<kbd>unpkg</kbd>](https://unpkg.com/browse/fusebit__add-on-sdk@latest/))
was missing the following properties:
1. createSettingsManager
1. createLifecycleManager
1. serializeState
1. deserializeState
1. getInputs

<details>
<summary>as well as these 8 other properties...</summary>
<p>completeWithSuccess, completeWithError, getSelfUrl, redirect, createFunction, deleteFunction, getFunctionDefinition, createFusebitFunctionFromExpress</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1a2752e1af8128848205edec1b6bd5e05b4f32ee
</p>

**Review by Hashiphoto — APPROVED** [review]
Looks good

File: types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts
Comment (peterblazejewicz) [inline types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1]:
```suggestion
import Sdk = require('@fusebit/add-on-sdk');
```

File: types/fusebit__add-on-sdk/index.d.ts
Comment (peterblazejewicz) [inline types/fusebit__add-on-sdk/index.d.ts:5]:
```suggestion
```
you also need ACK from this person, or convert to `@credit` mention. The DT package owners receive automated notification, so we expect explicit opt-in (ACK) (at least recently)

File: types/fusebit__add-on-sdk/index.d.ts
Comment (peterblazejewicz) [inline types/fusebit__add-on-sdk/index.d.ts:11]: do you folks need an enum? Could usual union be OK?
```ts
export type Method = 'GET' | 'POST' ...
```
if this is a real const, just create and use const:
https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums

**Review by andydam — APPROVED** [review]
Looks good to me

File: types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts
Comment (andrewrmiller) [reply types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1]: This suggestion results in:
```
Error: /hp/DefinitelyTyped/types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1:22
ERROR: 1:22   expect  TypeScript@4.9 compile error: 
Cannot find module '@fusebit/add-on-sdk' or its corresponding type declarations.
ERROR: 67:67  expect  TypeScript@4.9 compile error: 
Parameter 'client' implicitly has an 'any' type.
ERROR: 75:73  expect  TypeScript@4.9 compile error: 
Parameter 'url' implicitly has an 'any' type.
```

File: types/fusebit__add-on-sdk/index.d.ts
Comment (andrewrmiller) [reply types/fusebit__add-on-sdk/index.d.ts:5]: Andy has approved this PR.  Is that sufficient?

File: types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts
Comment (andrewrmiller) [reply types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1]: This change results in:
```
Error: /hp/DefinitelyTyped/types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1:22
ERROR: 1:22   expect  TypeScript@4.9 compile error: 
Cannot find module '@fusebit/add-on-sdk' or its corresponding type declarations.
ERROR: 67:67  expect  TypeScript@4.9 compile error: 
Parameter 'client' implicitly has an 'any' type.
ERROR: 75:73  expect  TypeScript@4.9 compile error: 
Parameter 'url' implicitly has an 'any' type.
```

Comment (andrewrmiller) [reply types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1]: This change results in:
```
Error: /hp/DefinitelyTyped/types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1:22
ERROR: 1:22   expect  TypeScript@4.9 compile error: 
Cannot find module '@fusebit/add-on-sdk' or its corresponding type declarations.
ERROR: 67:67  expect  TypeScript@4.9 compile error: 
Parameter 'client' implicitly has an 'any' type.
ERROR: 75:73  expect  TypeScript@4.9 compile error: 
Parameter 'url' implicitly has an 'any' type.
```

File: types/fusebit__add-on-sdk/index.d.ts
Comment (andrewrmiller) [reply types/fusebit__add-on-sdk/index.d.ts]: Union works.  Updated.

**Review by Hashiphoto — APPROVED** [review]

**Review by andydam — APPROVED** [review]
👍

File: types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts
Comment (andrewbranch) [reply types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1]: You need to add a `paths` entry to the tsconfig.json for it to work. This is the preferred way, but there are several packages that fail to do this already and it doesn’t matter a ton since this is in tests, so I won’t block on it, but if you want to fix it that would be great. You can look at `types/xmpp__client/tsconfig.json` for an example.

**Review by andrewbranch — APPROVED** [review]

File: types/fusebit__add-on-sdk/tsconfig.json
Comment (peterblazejewicz) [inline types/fusebit__add-on-sdk/tsconfig.json:22]:
```suggestion
    ],
    "paths": {
       "@fusebit/*": ["fusebit__*"]
    }
```

File: types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts
Comment (andrewrmiller) [reply types/fusebit__add-on-sdk/fusebit__add-on-sdk-tests.ts:1]: Fixed.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by andrewrmiller** [comment]
Ready to merge.

**Pull request merged**
